src/main/js/main_html5.js

```diff
@@ -1195,6 +1195,11 @@ require("../../../html5-common/js/utils/environment.js");
                       _video.currentTime);
           _.defer(raiseEndedEvent);
         }
+        if(OO.isSafari && !OO.isIos && !_video.ended && Math.round(_video.currentTime) === Math.round(_video.duration))
+        {
+          this.controller.notify(this.controller.EVENTS.SEEKED);
+          this.controller.notify(this.controller.EVENTS.ENDED);
+        }
```

Comment:
Can this all be merged with the section right above in some way?  I'm pretty sure they are very closely related; This seems like it should just require a small change to the if above instead of a whole new if